pkg/kp/rcstore/consul_store.go

```diff
@@ -121,6 +129,55 @@ func (s *consulStore) List() ([]fields.RC, error) {
 	return rcs, nil
 }
 
+func (s *consulStore) WatchNew(quit <-chan struct{}) (<-chan []fields.RC, <-chan error) {
+	outCh := make(chan []fields.RC)
+	errCh := make(chan error)
+	defer close(outCh)
```

Comment:
Does this immediately close them upon return from this function? I think you want to close them upon return from the goroutine instead.